subsys/net/lib/zperf/zperf_shell.c

```diff
@@ -1331,21 +1351,22 @@ SHELL_STATIC_SUBCMD_SET_CREATE(zperf_cmd_udp_download,
 SHELL_STATIC_SUBCMD_SET_CREATE(zperf_cmd_udp,
 	SHELL_CMD(upload, NULL,
 		  "[<options>] <dest ip> [<dest port> <duration> <packet size>[K] "
-							"<baud rate>[K|M]]\n"
+							"<baud rate>[K|M] <interface name>]\n"
 		  "<options>     command options (optional): [-S tos -a]\n"
 		  "<dest ip>     IP destination\n"
 		  "<dest port>   port destination\n"
 		  "<duration>    of the test in seconds\n"
 		  "<packet size> Size of the packet in byte or kilobyte "
 							"(with suffix K)\n"
 		  "<baud rate>   Baudrate in kilobyte or megabyte\n"
+		  "<interface>   Interface name\n"
```

Comment:
As the network interface name is an optional feature, it is better to use network interface index as a selector.

Reply:
iface idx is easier to change between different apps using the same driver.
eg. for the same iface(eth/wifi), its idx in APP A is 1 but in APP B is 2.
This case user is easy to write wrong idx.

So I think iface name is more robust and identifiable.

Reply:
True, that is a good point. It is also possible to change the interface name in the application but perhaps that is too far fetched example. Ok, we can go with the interface name then.